topdown/eval.go

```diff
@@ -909,20 +909,20 @@ func (e *eval) biunifyObjects(a, b ast.Object, b1, b2 *bindings, iter unifyItera
 		b = plugKeys(b, b2)
 	}
 
-	return e.biunifyObjectsRec(a, b, b1, b2, iter, a, 0)
+	return e.biunifyObjectsRec(a, b, b1, b2, iter, a, a.KeysIterator())
 }
 
-func (e *eval) biunifyObjectsRec(a, b ast.Object, b1, b2 *bindings, iter unifyIterator, keys ast.Object, idx int) error {
-	if idx == keys.Len() {
+func (e *eval) biunifyObjectsRec(a, b ast.Object, b1, b2 *bindings, iter unifyIterator, keys ast.Object, oki ast.ObjectKeysIterator) error {
+	key, moreKeysLeft := oki.Next() // Get next key from iterator.
+	if !moreKeysLeft {
 		return iter()
 	}
```

Comment:
[style nit] more is less is more 😉 
```suggestion
	key, more := oki.Next()
	if !more {
		return iter()
	}
```

Reply:
I ended up using that suggestion, and bundled it in to the other style-nits fixes this morning.